### PR TITLE
Annotate article

### PR DESCRIPTION
**Pull request by aysegulsari**
Since there is a connection problem while reading ontology concepts, I have used pickled concepts and the pickled concepts are not include all ontologies. Fyi.

### CODE REVIEW

**Review by 8hk — APPROVED** [review]
I just executed the python script and the values are inserted into the db. Everything seems correct and good for now. Thanks for your effortt. 🆙 👍

File: OntologyRetriever/annotate_abstract.py [hunk 1]
```diff
@@ -93,77 +101,150 @@ def retrieve_article_ids(search_term, max_article_limit):
...
+def retrieve_article(article_id):
+    response = requests.get(EntrezGetArticleRequest(article_id))
     if response.ok:
         xpars = xmltodict.parse(response.text)
         article = xpars["PubmedArticleSet"]["PubmedArticle"]["MedlineCitation"]["Article"]
         article_title = article["ArticleTitle"]
         journal_issn = article["Journal"]["ISSN"]["#text"]
         journal_name = article["Journal"]["Title"]
+        pubmed_link = "https://pubmed.ncbi.nlm.nih.gov/" + str(article_id) + "/"
+        abstract = article["Abstract"]["AbstractText"]
```
Comment (8hk) [inline OntologyRetriever/annotate_abstract.py]: we may need to add try catch block to catch any problem.

File: OntologyRetriever/annotate_abstract.py [hunk 2]
```diff
@@ -93,77 +101,150 @@ def retrieve_article_ids(search_term, max_article_limit):
...
+                        # print("ONTOLOGY CONCEPT: " + c.pref_label + " POSITION START:" + str(position['start']) + " POSITION END:" + str(position['end']) + "\n")
+                        # print("Article with id: " + retrieved_article_ids[id] + " has ontolgy concept: " + c.id + " (synonyms=" + c.pref_label + ")")
+                        annotation_object = create_annotation_object(annotation_counter, article, c, position)
+                        if article.pm_id not in annotated_article_ids:
+                            annotated_article_ids.append(article.pm_id)
+                        print(annotation_object)
+                        annotation_counter = annotation_counter + 1
+                        annotation_list.append(annotation_object)
+                    print("\n--------------------------------------------")
+    return convert_to_json_abjects(annotated_article_ids)
```
Comment (8hk) [inline OntologyRetriever/annotate_abstract.py]: we may also need to add try except block here to catch problems

File: OntologyRetriever/annotate_abstract.py [hunk 3]
```diff
@@ -16,154 +10,280 @@
...
-        print("\tThis article id could not be retrieved.")
+        if response.ok:
+            xpars = xmltodict.parse(response.text)
+            article_count = xpars['eSearchResult']['Count']
+            returned_article_id_count = int(xpars['eSearchResult']['RetMax'])
+            print("From " + article_count + " articles " + str(
+                returned_article_id_count) + " article ids are retrieved")
+            return xpars['eSearchResult']['IdList']['Id']
+        else:
+            print("\tThis article id could not be retrieved.")
```
Comment (heobu) [inline OntologyRetriever/annotate_abstract.py:114]: can we add the article id to this log maybe?

**Review by heobu — APPROVED** [review]

**Pull request merged**